test/jdk/javax/swing/JEditorPane/bug4694598.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2002, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @bug 4694598
+ * @key headful
+ * @summary JEditor pane throws NullPointerException on mouse movement.
+ * @run main bug4694598
+ */
+
+import java.awt.AWTException;
+import java.awt.Robot;
+import java.io.IOException;
+import java.io.Writer;
+import java.lang.reflect.InvocationTargetException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import javax.swing.JEditorPane;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+
+public class bug4694598 {
+    JFrame frame;
+    volatile int bottom;
+    final Path frameContentFile = Path.of("FrameContent.html");
+
+    public void setupGUI() {
+        frame = new JFrame("Test 4694598");
+        JEditorPane jep = new JEditorPane();
+        jep.setEditable(false);
+        frame.getContentPane().add(jep);
+        frame.setLocation(50, 50);
+        frame.setSize(400, 400);
+
+        String frameContentString = "<HTML><BODY>\n" +
+                "    Frame content.\n" +
+                "</BODY></HTML>";
+        try (Writer writer = Files.newBufferedWriter(frameContentFile)) {
+            writer.write(frameContentString);
+        } catch (IOException ioe){
+            throw new RuntimeException("Could not create html file to embed", ioe);
+        }
```

Comment:
Move creating the file to `main` method before setting up GUI and let `IOException` escape from main.

Reply:
This is try-with-resources so if i will to it in main i will have to add synchronizing and closing of writer which is a strange trade-off so i would have to do try block anyways.

Reply:
Why would you need synchronisation?

```java
        try (Writer writer = Files.newBufferedWriter(frameContentFile)) {
            writer.write(frameContentString);
        }
```
would write out the contents of the file and close the file handle. In setupGui, you would still use `frameContentFile` object only which is `final` and immutable, therefore it's thread-safe.

Reply:
And that's pretty much what i do here - except for the exception handling. I just do not see a reason to move it to main, it makes no difference except now the HTML creation will be split and harder to understand.